pil-analyzer/src/evaluator.rs

```diff
@@ -461,6 +471,20 @@ impl<'a> Definitions<'a> {
                         Value::TypeConstructor(&variant.name).into()
                     }
                 }
+                Some(FunctionValueDefinition::TraitFunction(_, _)) => {
+                    let type_arg = type_args.clone().unwrap();
```

Comment:
I think it would be nice to have an assertion that `type_arg` is concrete.

Reply:
although actually I think this should always be the case here, right? So we could put it at the start of the function.

Reply:
I mean we could do that at the call site, but I yeah, probably it's fine. Then let's not do that at all.

Reply:
There is already one, but inside the traits_resolver, before the unifier.
https://github.com/powdr-labs/powdr/blob/0af8e32a68aca886660f8d421305a2f938f8888c/pil-analyzer/src/traits_resolver.rs#L68